src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -293,7 +295,12 @@ private void putInternal(Model model, ActionListener<IndexResponse> listener, Do
                     put(KNNConstants.MODEL_ERROR, modelMetadata.getError());
                     put(KNNConstants.MODEL_NODE_ASSIGNMENT, modelMetadata.getNodeAssignment());
                     put(KNNConstants.VECTOR_DATA_TYPE_FIELD, modelMetadata.getVectorDataType());
-
+                    if (Mode.isConfigured(modelMetadata.getMode())) {
```

Comment:
Wonder if `isConfigured` method is really required.
Can we just simply call `put(KNNConstants.MODE_PARAMETER, modelMetadata.getMode())`?

Reply:
Do you think its okay to write null here? I was thinking we could avoid doing it.

Reply:
I think it should be okay. Anyway, we are not checking if the key exist while reading it but checking the value is null or not.